src/main/java/me/qoomon/gradle/gitversioning/GitVersioningPlugin.java

```diff
@@ -14,7 +14,13 @@ public void apply(@Nonnull Project project) {
 
         project.getExtensions().create("gitVersioning", GitVersioningPluginExtension.class, project);
 
-        project.getAllprojects().forEach(it -> it.getTasks().create("version", VersionTask.class));
+        // Only Register version task for current project so the plugin may be applied
+        // on
+        // sub projects. It would be nice to only define this once for all project in
+        // a multi-module project, however there are a few other considerations that
+        // need
```

Comment:
I think we need to describe the problem to solve ("...other considerations that need to be made...").

Reply:
The current change scopes the plugin to the project in which is registered. This is the common pattern for most plugins. There are a few paths that could be followed.

1. Leave as is where a version task is registered everywhere. I personally don't see the value in only registering part of the plugin. This registers the task, but does not apply the configuration aspects. If this is choose, the task should be registered conditionally and only afterEvaulate has completed. This would still allow the plug to be registered on other projects, which today it fails
2. Take the change as is only register the plugin and it's tasks in the project requested. This is the pattern most plugins follow. It's the safest and easiest to get correct.
3. Register and apply not only the version task, but also the versioning semantics to every project when in a multi-project. The challenge with this is that since the versioning happens during the configuration phase ordering is very important and getting this right will be difficult

Reply:
I would consider this a bug and not an enhancement, since as the plugin is currently authored it is unusable in a multi-project structure.